onnx/defs/traditionalml/defs.cc

```diff
@@ -413,6 +391,21 @@ ONNX_ML_OPERATOR_SET_SCHEMA(
                 " must be the same in the LabelEncoder.");
           }
 
+          auto default_attr = ctx.getAttribute("default_tensor");
```

Comment:
The new test just ensures the consistency of the operator when default_tensor is set. According to the description of the PR, it was failing in any other case. No unit test is failing, so I assume it was not tested. Is it possible to fix the behaviour and add unit test to check it is doing as expected?

Reply:
It's not quite that. When implementing the operator in ORT, I was finding the type inference fails even if I have only specified one of the default_* attributes. There seems to be to do with the fact that the default_* attributes tend to have values specified since they are optional.

Reply:
Some failure cases are checked [here](https://github.com/onnx/onnx/blob/a93ec80919a4370a4b160eef811488d569487cfc/onnx/test/shape_inference_test.py#L5719-L5785). Is there something you think is missing?

Reply:
All tests involve `default_tensor`, there is no test to check that a default value is not defined or defined with an unexpected attribute such as (value_int64s, default_floats). A test like that would have failed with your change.

Reply:
> there is no test to check that a default value is not defined

I don't think that is strictly speaking necessary, since the default value has a fallback (e.g. -1 for int64_t) specified in the specification. This is why it is not tested in the previous implementation I believe.


> defined with an unexpected attribute such as (value_int64s, default_floats)

Sure, I can add this.

Reply:
Just to clarify Aditya's point (in answer to Xavier's question): since the op-spec specifies a default-value for the various attributes, we need to be careful when flagging an error here: we cannot do that unless we can distinguish whether an attribute-value was explicitly specified in the node, or whether it is getting a default-value from the op-spec. (We should not be flagging an error when we get a default-value from the op-spec.) At the root of this, we have the issue that shape-inference works with the `InferenceContext` abstract interface, which exposes the `getAttribute` method. If we want to do such error-checking, we need to extend the interface to distinguish between the two cases mentioned above.)

So, undoing this check makes sense to me. My two cents.

Reply:
Explained much better than me! Thanks